src/libstd/collections/hash/table.rs

```diff
@@ -897,8 +898,9 @@ unsafe impl<K: Send, V: Send> Send for IntoIter<K, V> {}
 
 /// Iterator over the entries in a table, clearing the table.
 pub struct Drain<'a, K: 'a, V: 'a> {
-    table: &'a mut RawTable<K, V>,
+    table: Shared<RawTable<K, V>>,
```

Comment:
This seems curious because today it's implemented in seeming "safe" code, but that didn't get us the right variance by default?